auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -834,93 +809,1294 @@ public void simple() {
-        + "(TypeAdapter<List<String>>) gson.getAdapter(TypeToken.getParameterized(List.class, "
-        + "String.class));\n"
+        + "              TypeAdapter<List<? extends String>> list__wildcard__string_adapter = this.list__wildcard__string_adapter;\n"
+        + "              if (list__wildcard__string_adapter == null) {\n"
+        + "                this.list__wildcard__string_adapter = list__wildcard__string_adapter = (TypeAdapter<List<? extends String>>) gson.getAdapter(TypeToken.getParameterized(List.class, WildcardUtil.subtypeOf(String.class)));\n"
         + "              }\n"
-        + "              j = list__string_adapter.read(jsonReader);\n"
+        + "              builder.j(list__wildcard__string_adapter.read(jsonReader));\n"
+        + "              continue;\n"
+        + "            }\n"
+        + "            if (realFieldNames.get(\"o\").equals(_name)) {\n"
+        + "              TypeAdapter<Map<String, Map<String, Map<String, Map<String, Map<String, ? extends String>>>>>> map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter = this.map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter;\n"
+        + "              if (map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter == null) {\n"
+        + "                this.map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter = map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter = (TypeAdapter<Map<String, Map<String, Map<String, Map<String, Map<String, ? extends String>>>>>>) gson.getAdapter(TypeToken.getParameterized(Map.class, String.class, TypeToken.getParameterized(Map.class, String.class, TypeToken.getParameterized(Map.class, String.class, TypeToken.getParameterized(Map.class, String.class, TypeToken.getParameterized(Map.class, String.class, WildcardUtil.subtypeOf(String.class)).getType()).getType()).getType()).getType()));\n"
+        + "              }\n"
+        + "              builder.o(map__string_map__string_map__string_map__string_map__string_wildcard__string_adapter.read(jsonReader));\n"
         + "              continue;\n"
         + "            }\n"
         + "            jsonReader.skipValue();\n"
         + "          }\n"
         + "        }\n"
         + "      }\n"
         + "      jsonReader.endObject();\n"
-        + "      return new AutoValue_Test(a, b, c, d, e, f, g, h, i, j);\n"
+        + "      return builder.build();\n"
+        + "    }\n"
+        + "  }\n"
+        + "}"
+    );
+
+    assertAbout(javaSources())
+        .that(Arrays.asList(nullable, source))
+        .processedWith(new AutoValueProcessor(Lists.newArrayList(new AutoValueGsonExtension())))
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expected);
+  }
+
+  @Test
+  public void simpleWithAnnotatedBuilder() {
```

Comment:
If you wanted to just move this into the sample project as a code test there rather than this, I'd be in favor of starting that pattern with this PR. Seems way more maintainable

Reply:
Same with the other new tests below